ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -277,6 +298,21 @@ public Integer getValue() {
   }
 
   /**
+   * Initializes a {@link Gauge} metric for the maxPutChunkSize. This value is a constant for a running router
+   * instance, but can change between invocations. This metric helps maintain historical information.
+   * @param maxPutChunkSize the value to initialize this metric with.
+   */
+  public void initializeMaxPutChunkSizeMetric(final int maxPutChunkSize) {
+    maxPutChunkSizeGauge = new Gauge<Integer>() {
+      @Override
+      public Integer getValue() {
+        return maxPutChunkSize;
+      }
+    };
+    metricRegistry.register(MetricRegistry.name(NonBlockingRouter.class, "MaxPutChunkSizeBytes"), maxPutChunkSizeGauge);
```

Comment:
This value can change between invocations of the router, if we play around with chunk sizes. As mentioned in the description: though this would not change for a running router, this metric would help maintain historical information if the chunk sizes are changed. We could probably get rid of this eventually, but I think it will be useful at least in the interim. This should be harmless as the only cost is that of the gauge itself - no computation costs. Is there a better way to track this information?